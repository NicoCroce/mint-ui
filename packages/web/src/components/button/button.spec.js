import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import "jest-styled-components";

import Button from "./button";

describe("Button", () => {
	test("renders button", () => {
		const { asFragment } = render(<Button>Test</Button>);
		expect(asFragment()).toMatchSnapshot();
	});
	test("captures button click", done => {
		function handleClick() {
			done();
		}
		const { getByTestId } = render(
			<Button data-testid="capture-click" onClick={handleClick}>
				Capture click
			</Button>
		);
		const node = getByTestId("capture-click");
		fireEvent.click(node);
	});
});

afterEach(cleanup);
