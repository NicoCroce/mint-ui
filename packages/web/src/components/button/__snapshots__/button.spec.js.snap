// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 16px;
  font-weight: 500;
  cursor: pointer;
  text-align: center;
  line-height: 1.25;
  padding: 14px 24px;
  display: inline-block;
  background: #00a465;
  border-radius: 4px;
  border: 1px solid transparent;
  color: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  padding: 10px 16px;
}

.c0:hover {
  background: #0fb877;
}

.c0:active {
  color: #fff;
  background: #078a58;
}

<button
    class="c0 tm-button- "
    type="button"
  >
    <span
      style="vertical-align: middle; margin-left: 0px;"
    >
      Test
    </span>
  </button>
</DocumentFragment>
`;
